src/main/java/ita/softserve/course_evaluation/service/impl/PermissionServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package ita.softserve.course_evaluation.service.impl;
+
+import ita.softserve.course_evaluation.dto.PermissionDto;
+import ita.softserve.course_evaluation.dto.PermissionDtoMapper;
+import ita.softserve.course_evaluation.entity.Permission;
+import ita.softserve.course_evaluation.repository.PermissionRepository;
+import ita.softserve.course_evaluation.service.PermissionService;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class PermissionServiceImpl implements PermissionService {
+
+    private final PermissionRepository permissionRepository;
+
+    public PermissionServiceImpl(PermissionRepository permissionRepository) {
+        this.permissionRepository = permissionRepository;
+    }
+
+    @Override
+    public Permission create(PermissionDto dto) {
+        String name = dto.getPermissionName();
+        if (findByName(name).isPresent()) {
```

Comment:
```findByName(name).isPresent()``` probably is modern approach :)

Reply:
Also - I suggest using the repo`s method instead of local method

Reply:
If I create or update permission.
I want to throw the exception when the permission with name is already exists in database. 
What is the better way to do it? @Forfend

Reply:
The better way is to not manage roles\permissions vie REST API at all. It`s better to do it via sql scripts or liquibase migrations
@VovaProtsiv